core/controllers/acl_decorators_test.py

```diff
@@ -1216,6 +1216,52 @@ def test_normal_user_can_manage_preferences(self):
         self.logout()
 
 
+class ManageOwnProfileTests(test_utils.GenericTestBase):
+    """Tests for decorator can_manage_own_profile."""
+
+    banned_user = 'banneduser'
+    banned_user_email = 'banned@example.com'
+    username = 'user'
+    user_email = 'user@example.com'
+
+    class MockHandler(base.BaseHandler):
+        GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+        @acl_decorators.can_manage_own_profile
+        def get(self):
+            return self.render_json({'success': 1})
+
+    def setUp(self):
+        super(ManageOwnProfileTests, self).setUp()
+        self.signup(self.banned_user_email, self.banned_user)
+        self.signup(self.user_email, self.username)
+        self.set_banned_users([self.banned_user])
+        self.mock_testapp = webtest.TestApp(webapp2.WSGIApplication(
+            [webapp2.Route('/mock/', self.MockHandler)],
+            debug=feconf.DEBUG,
+        ))
+
+    def test_banned_user_cannot_update_profile_preferences(self):
+        self.login(self.banned_user_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            self.get_json('/mock/', expected_status_int=401)
+        self.logout()
+
+    def test_guest_cannot_update_profile_preferences(self):
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.get_json('/mock/', expected_status_int=401)
+        self.assertEqual(
+            response['error'],
+            'You must be logged in to access this resource.')
+
+    def test_normal_user_can_manage_profile_preferences(self):
```

Comment:
Suggest being more specific than 'normal user' now that we have several different types of users to consider noq.

Reply:
Does `normal_full_user` sounds good?
```python
def test_normal_full_user_can_manage_profile_preferences(self):
        self.login(self.user_email)
        with self.swap(self, 'testapp', self.mock_testapp):
            response = self.get_json('/mock/')
        self.assertEqual(response['success'], 1)
        self.logout()
```
Also should I go ahead and make this name change in rest of the functions too, because elsewhere also (at about 18 other places in this test file) normal_user is referring to the normal "full user" only.

Reply:
Aren't we also referring to 'full account' in some places? I'm concerned about mismatching terminology. I'm not a huge fan of 'full user' but I can't think of anything clearer, so let's go with that. You should update everywhere that refers to accounts inconsistently to ensure we consistently use the same term for the same thing.

Reply:
I haven't used the term `full account` anywhere in the codebase, only in the design doc I have used it interchangeably with `full user` sometimes.
Code base usually refers to users rather than account, hence I have not used it. To be absolutely sure I have checked across all open branches too and will keep that in mind in future also.

Reply:
Sounds good--thanks.